### DISCUSSION TITLE
I need help with Dapp development.

### DISCUSSION THREAD

**Discussion by MetaAhn** [WalletConnect/walletconnect-monorepo#2377] [Developer Support] [answered]
I am developing a Dapp using React, Web3Modal SDK (v2), and Wagmi.

First question:
My Dapp should be accessible on mobile devices as well.
When accessing my Dapp on a mobile device and the mobile (Wi-Fi) network changes, the interaction with the MetaMask app gets disrupted.
Can you share a solution to resolve this issue or a way to detect changes in the mobile (Wi-Fi) network?

Second question:
There are instances where transactions in the MetaMask application, which interact with my Dapp, become pending.
Is it possible to use the Web3Modal SDK to check for pending transactions? If not, are there any suitable alternatives?

**Comment by chendinh** [comment]
First question:
-> you can check the status of your wifi, Init again with walletconnect and Resubsribe all sockets again. the interaction will work again.
Second question:
-> i dont understand what do you want exactly. You can use contract ankr https://www.ankr.com/rpc/advanced-api 
to check status of transaction

**Reply by MetaAhn** [reply]
Can you please provide a reference for the first question?

The example given in the second question refers to a situation where transactions are pending in the MetaMask application, as shown in the following image. It takes a significant amount of time to connect a wallet holding pending transactions to my Dapp and proceed with transaction signing/sending.
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/105259320/b2edc524-fe90-4c6d-bf55-f2f5b30c3614)
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/105259320/dc05b2bf-b21e-47cb-aa6b-747c5c8fbfa3)

**Comment by finessevanes** [comment, marked as answer]
1. You can check if the client exists: https://github.com/WalletConnect/web3modal-examples/blob/ff3c55b9555c38f3656494608bd63998a9964ef7/nextjs-wagmi/src/pages/_app.jsx#L29
2. Use wagmi hooks: https://wagmi.sh/react/hooks/useWatchPendingTransactions
